library/core/networking/src/main/java/org/quiltmc/qsl/networking/api/client/ClientPlayNetworking.java

```diff
@@ -0,0 +1,261 @@
+	 *
+	 * @param channelName the id of the channel
+	 * @return false if a handler is already registered to the channel
+	 * @throws IllegalStateException if the client is not connected to a server
+	 * @see ClientPlayConnectionEvents#INIT
+	 */
+	public static boolean registerReceiver(Identifier channelName, PlayChannelReceiver channelHandler) {
+		final ClientPlayNetworkAddon addon = ClientNetworkingImpl.getClientPlayAddon();
+
+		if (addon != null) {
+			return addon.registerChannel(channelName, channelHandler);
+		}
+
+		throw new IllegalStateException("Cannot register receiver while not in game!");
+	}
+
+	/**
+	 * Removes the handler of a channel.
+	 *
+	 * <p>The {@code channelName} is guaranteed not to have a handler after this call.
+	 *
+	 * @param channelName the id of the channel
+	 * @return the previous handler, or {@code null} if no handler was bound to the channel
+	 * @throws IllegalStateException if the client is not connected to a server
+	 */
+	@Nullable
+	public static ClientPlayNetworking.PlayChannelReceiver unregisterReceiver(Identifier channelName) throws IllegalStateException {
+		final ClientPlayNetworkAddon addon = ClientNetworkingImpl.getClientPlayAddon();
+
+		if (addon != null) {
+			return addon.unregisterChannel(channelName);
+		}
+
+		throw new IllegalStateException("Cannot unregister receiver while not in game!");
+	}
+
+	/**
+	 * Gets all the channel names that the client can receive packets on.
+	 *
+	 * @return All the channel names that the client can receive packets on
+	 * @throws IllegalStateException if the client is not connected to a server
+	 */
+	public static Set<Identifier> getReceived() throws IllegalStateException {
+		final ClientPlayNetworkAddon addon = ClientNetworkingImpl.getClientPlayAddon();
+
+		if (addon != null) {
+			return addon.getReceivableChannels();
+		}
+
+		throw new IllegalStateException("Cannot get a list of channels the client can receive packets on while not in game!");
+	}
+
+	/**
+	 * Gets all channel names that the connected server declared the ability to receive a packets on.
+	 *
+	 * @return All the channel names the connected server declared the ability to receive a packets on
+	 * @throws IllegalStateException if the client is not connected to a server
+	 */
+	public static Set<Identifier> getSendable() throws IllegalStateException {
+		final ClientPlayNetworkAddon addon = ClientNetworkingImpl.getClientPlayAddon();
+
+		if (addon != null) {
+			return addon.getSendableChannels();
+		}
+
+		throw new IllegalStateException("Cannot get a list of channels the server can receive packets on while not in game!");
+	}
+
+	/**
+	 * Checks if the connected server declared the ability to receive a packet on a specified channel name.
+	 *
+	 * @param channelName the channel name
+	 * @return True if the connected server has declared the ability to receive a packet on the specified channel.
```

Comment:
```suggestion
	 * @return {@code true} if the connected server has declared the ability to receive a packet on the specified channel, otherwise {@code false}
```